### DISCUSSION TITLE
How to best see history in HA

### DISCUSSION THREAD

**Discussion by rhatguy** [PeteRager/lennoxs30#190] [General]
I've recently added a Lennox EL18XCV variable speed AC unit with a S30 thermostat on the main level in my house.  In HA on my old thermostat I could see when the unit was running in the graph that is displayed when clicking on the unit in HA (see my basement AC below).  This new system keeps the temperature so constant, I can go for an entire day without the purple line on the graph (my setpoint) changing.  This makes it challenging to know when the unit has run.  You can see in the attached main level graph where we changed the setpoint yesterday and then we can see the blue shaded area under the purple setpoint line indicating that the unit was running, but since then I can't see when the unit has run, since the purple setpoint line is on the bottom of the graph.  This seems like more of a graph scaling problem in Home Assistant, but just curious if others here have similar issues and how you handle?

![new_thermostat](https://user-images.githubusercontent.com/11964411/185209823-1e5ff388-bd9d-460f-a454-8c91fe9b2b87.JPG)
![old_thermostat](https://user-images.githubusercontent.com/11964411/185209826-bd3fc9c5-17af-4dd2-bafb-bb2111d99eaa.JPG)

**Comment by PeteRager** [comment]
It should work as you expect in picture 2.  Here's a screen shot of this history from my AC.

![im](https://user-images.githubusercontent.com/76050312/185241021-e9a367b8-a230-4deb-8a47-a86f60dd54c0.jpg)

So there are a couple of possibilities.

a) The temperature is holding at 75 and the HVAC system is not doing anything.  This would be "everything is working as it is supposed to", and there is no HVAC demand.  There is an attribute of the climate object called hvac_action.  It may be worthwhile pulling this attribute into a sensor template and recording it - it has values like "cooling", "heating", "dehumidifying", etc.  I'd also check that the Lennox Panel and HA are showing the same values for air temperature, humidity, and the setpoint.

b) The integration does not have connection to the thermostat or the thermostat has stopped sending updates.  This would be "something is not working as it is supposed to".  The first thing to look at here are the message counters.  There should be an entity called lennoxs30.state, it has a bunch of attributes, one of them is MessageCount - this number should be increasing, for a local connection it should update at least once a minute.  If errors are encountered these counters will increase.  Also look in the home assistant log for any errors.

![cap](https://user-images.githubusercontent.com/76050312/185242627-b378da76-4722-4d46-9bea-b0de50408aee.png)

What versions of HA and the integration are you running?  Cloud or local connection?

If you think it's not working, the next step would be to enable message logging.  The readme describes how to do this.

**Reply by rhatguy** [reply]
I think everything is working as expected.  I'm able to change setpoints and such and if I zoom out far enough on the history to where there were changes, like you did in your screenshot the graph looks similar to yours.  Its definitely getting data from the thermostat as you can see some shading on my main level graph.

My issue is that if I don't change my thermostat setpoint, the variable speed AC unit keeps the temperature literally constant (I.E. exactly 75 degrees all the time).  If it goes long enough like that the history graph for the last day or so changes the scale and then you can't see the blue highlighted areas where the unit was running.  If I go to the history component and zoom out far enough like you did in your screenshot I can see when it was running, but if I click on the unit in the simple thermostat card or view a time period with no change in temperature, the scale of the graph changes and I can't see the blue "running" lines anymore.

**Reply by PeteRager** [reply]
Ok, got it.  What you can do is put the climate.hvac_action into a sensor template and then graph that.

Something like this
```
sensor:
  - platform: template
    sensors:
      hvac_sys_uptime:
        unit_of_measurement: sec
        value_template: >-
          {{ state_attr('lennoxs30.state','sysUpTime') }}
      air_conditioner_hvac_action:
        value_template: >-
          {{ state_attr('climate.air_conditioner','hvac_action') }}
      air_conditioner_humidity_action:
        value_template: >-
          {{ state_attr('climate.air_conditioner','humOperation') }}
 ```

and then add them to your recorder config.
